source/codegen/templates/service_helpers.mako

```diff
@@ -158,13 +158,21 @@ ${initialize_standard_input_param(function_name, parameter)}\
   request_snippet = f'request->{field_name}()'
   c_type = parameter['type']
   c_type_pointer = c_type.replace('[]','*')
+  c_type_underlying_type = common_helpers.get_underlying_type_name(c_type)
 %>\
 % if c_type == 'ViConstString':
       ${c_type} ${parameter_name} = ${request_snippet}.c_str();\
 % elif c_type == 'ViString' or c_type == 'ViRsrc':
       ${c_type} ${parameter_name} = (${c_type})${request_snippet}.c_str();\
 % elif c_type == 'ViInt8[]' or c_type == 'ViChar[]':
       ${c_type_pointer} ${parameter_name} = (${c_type[:-2]}*)${request_snippet}.c_str();\
+% elif c_type == 'ViBoolean[]':
+      auto ${parameter_name}_data = ${request_snippet}.data();
+      int ${parameter_name}_size = request->${field_name}_size();
+      std::vector<${c_type_underlying_type}> ${parameter_name};
+      for (int i=0; i< ${parameter_name}_size; i++) {
+          ${parameter_name}.push_back(${c_type_underlying_type}(${parameter_name}_data[i]));
+      }
```

Comment:
It's usually better to use C++ iterators to copy vectors rather than using for loops. It also prevents us from defining `i` multiple times in the same function.
```suggestion
      auto ${parameter_name}_request = ${request_snippet};
      std::vector<${c_type_underlying_type}> ${parameter_name};
      std::transform(${parameter_name}_request.begin(), ${parameter_name}_request.end(), ${parameter_name}.begin(), [](auto x) { return (${c_type_underlying_type})x; });
```
I think this should work, but you might need to play around with this to get it to compile. I'm assuming that we can use auto params in lambdas.

Reply:
Had to change third parameter to use `std::back_inserter(${parameter_name})` instead of `${parameter_name}.begin()`, but that worked, thanks. Didn't know `transform` function for vectors.